pkg/services/settings/settingsprovider/keyval.go

```diff
@@ -0,0 +1,45 @@
+package settingsprovider
+
+import (
+	"strconv"
+	"time"
+)
+
+type keyValue struct {
+	key   string
```

Comment:
Go generics are coming, but not sure how that could help here yet. So, I decided to keep using strings as the internal representation for keys and values, which is basically what's currently being used by the INI library, and still valid since no matter the origin (INI file, CLI args, envvars, database) every settings value can be represented as a string.

Reply:
This seems reasonable to me.